src/cmd-init

```diff
@@ -126,13 +181,59 @@ mkdir -p src
             (set +x; cd config && echo -n "Config commit: " && git describe --tags --always --abbrev=42)
             ;;
      esac
-     manifest="config/manifest.yaml"
-     if [[ ! -f "${manifest}" ]]; then
-         echo 1>&2 "Failed to find src/${manifest}"
-         fatal "If using a custom configuration, be sure it has a manifest.yaml."
-     fi
  fi)
 
+# Setup the locally stored init config for a given variant
+setup_variant() {
+    if [[ "${#}" -ne 1 ]]; then
+        echo 1>&2 "Error: This function must be called with the variant name"
+    fi
+    local -r variant="${1}"
+    local -r variant_file="src/config/variant_${variant}"
+
+    if [[ ! -f "${variant_file}" ]]; then
+        echo 1>&2 "Could not find the config for the '${variant} variant (${variant_file})"
+        fatal "If you are using a custom configuration, be sure it has a manifest.yaml & image.yaml."
+    fi
+
+    osver="$(head -1 "${variant_file}")"
+    manifest="src/config/manifest-${osver}.yaml"
+    image="src/config/image-${osver}.yaml"
+    if [[ ! -f "${manifest}" ]] || [[ ! -f "${image}" ]]; then
+        fatal "Could not find the manifests (${manifest} & ${image}) for the '${variant} variant"
+    fi
+
+    echo "Using variant: ${variant} (${osver})"
+    cat > "src/config.json" <<EOF
```

Comment:
I think this should go in say `tmp/config.json`.  Conceptually it's a generated/derived file.  Today for example a `cosa clean --all` should clean up everything, but this file is in a new place it doesn't know how to clean.

Reply:
Not sure I follow. We don't want that file to be cleaned up by `cosa clean -all` as it will change the variant being built as it will restore the default variant.

Reply:
Right now we can only specify the variant to build during `cosa init` (or you can change it manually but we don't expose it).

Reply:
I could make this a separated command `cosa set-variant` that folks would have to remember setting when they cleanup.

Reply:
OK, fair enough!